supervisor: add clearnet-ibd finished handling

0xB10C opened the pull request:
When the IBD is finished and the clearnet IBD key is set to 1 (IBD over clearnet) it is set back to 0 (IBD over Tor).

Stadicus commented on tools/bbbsupervisor/bbbsupervisor.go:
```diff
@@ -486,23 +490,46 @@ func handleBitcoindIDB(event watcherEvent, pState *supervisorState) error {
+// unsetClearnetIDB unsets (0 - download blocks over Tor) the ibdClearnetRedisKey if set.
+// The key can only be set back to 1 (download blocks over clearnet) via RPC.
+func unsetClearnetIDB() (err error) {
+	const ibdClearnetRedisKey string = "bitcoin:ibd-clearnet"
```
Redis key is `bitcoind:ibd-clearnet`

Stadicus commented on tools/bbbsupervisor/bbbsupervisor.go:
```diff
@@ -486,23 +490,46 @@ func handleBitcoindIDB(event watcherEvent, pState *supervisorState) error {
+	}
+	if isIBDClearnet == 1 {
+		log.Printf("IDB finished. Setting %s to %d.\n", ibdClearnetRedisKey, 0)
+		err = setRedisValue(ibdClearnetRedisKey, "0")
```
My bad, it's not enough that the Redis key is set to `0`, also the bitcoind config file must be changed and the service needs to be restarted. The supervisor should call the following command, which also sets the Redis value back to zero.
```
bbb-config.sh disable bitcoin_ibd_clearnet
```
Sorry about the sloppy spec... 😶

0xB10C commented:
Changed redis key to `bitcoind:ibd-clearnet`, added a `set to 0` if `ìbd==false` on supervisor startup and replaced `setRedisValue` with a `setBBBConfigValue`.

Stadicus commented on tools/bbbsupervisor/bbbsupervisor.go:
```diff
@@ -462,47 +466,75 @@ func handleMiddlewareNoBitcoindConnectivity(event watcherEvent, pState *supervis
+	}
+	if isIBDClearnet == 1 {
+		log.Printf("IDB finished. Setting %s to %d.\n", ibdClearnetRedisKey, 0)
+		err := setBBBConfigValue("disable", "bitcoin_ibd_clearnet")
```
IIUC, the function `setBBBConfigValue` would call the following script
```
bbb-config.sh set disable bitcoin_ibd_clearnet
```
This would be wrong, as it should be `bbb-config.sh disable bitcoin_ibd_clearnet`. Thinking of it, I could move the command from "enable/disable" section into "set/get". Then this line should be 
```
err := setBBBConfigValue("bitcoin_ibd_clearnet", "false")
```
Should I adjust the bash script?

0xB10C replied:
Yeah you are totally right.. I should check closer.. 

Ok. Let's move it from enable/disable to set/get. I'll fix it here.

0xB10C replied:
If changed, the middleware RPC for that added in 5b6029528e77552dff334555b22f14a131afb7bd needs to be updated as well.

Stadicus replied:
See draft pull-request https://github.com/digitalbitbox/bitbox-base/pull/190 to change bash script

0xB10C commented:
Now must be merged with #193 and #187.
